addons/mail/static/src/web/activity/activity_service.js

```diff
@@ -51,24 +58,60 @@ export class ActivityService {
 
     /**
      * @param {import("./activity_model").Data} data
+     * @param {Object} [param1]
+     * @param {boolean} param1.broadcast
      * @returns {import("./activity_model").Activity}
      */
-    insert(data) {
+    insert(data, { broadcast = true } = {}) {
         const activity = this.store.activities[data.id] ?? new Activity(this.store, data.id);
         if (data.request_partner_id) {
             data.request_partner_id = data.request_partner_id[0];
         }
         assignDefined(activity, data);
+        if (broadcast) {
+            this.broadcastChannel.postMessage({
+                type: "insert",
+                payload: this._serialize(activity),
+            });
+        }
         return activity;
     }
 
-    delete(activity) {
+    delete(activity, { broadcast = true } = {}) {
         delete this.store.activities[activity.id];
+        if (broadcast) {
+            this.broadcastChannel.postMessage({ type: "delete", payload: { id: activity.id } });
+        }
+    }
+
+    _onBroadcastChannelMessage({ data }) {
+        switch (data.type) {
+            case "insert":
+                this.insert(data.payload, { broadcast: false });
+                break;
+            case "delete":
+                this.delete(data.payload, { broadcast: false });
+                break;
+            case "reload chatter": {
+                const thread = this.env.services["mail.thread"].getThread(
+                    data.payload.resModel,
+                    data.payload.resId
+                );
+                this.env.services["mail.thread"].fetchNewMessages(thread);
```

Comment:
Explicitly use the service passed in constructor.
I'm doing a PR to clean them, so don't worry too much on trying to fix the problem by being inconsistent